consul/prepared_query_endpoint.go

```diff
@@ -562,6 +572,18 @@ func tagFilter(tags []string, nodes structs.CheckServiceNodes) structs.CheckServ
 	return nodes[:n]
 }
 
+// nodeMetaFilter returns a list of the nodes who satisfy the given metadata filters. Nodes
+// must have ALL the given tags.
+func nodeMetaFilter(filters map[string]string, nodes structs.CheckServiceNodes) structs.CheckServiceNodes {
```

Comment:
Yeah it would depend on the case. I realized that `CheckServiceNodes` isn't a slice of pointers, so we might want to change that eventually so we don't actually copy the structure around. That would kind of outside the scope of this change, so we should probably add a TODO comment and address that later. I'm ok with this as-is, if it pops up in a profile later we can address it later.